src/libraries/Microsoft.Extensions.Configuration/ref/Microsoft.Extensions.Configuration.cs

```diff
@@ -35,6 +35,7 @@ public partial class ConfigurationBuilder : Microsoft.Extensions.Configuration.I
         public ConfigurationBuilder() { }
         public System.Collections.Generic.IDictionary<string, object> Properties { get { throw null; } }
         public System.Collections.Generic.IList<Microsoft.Extensions.Configuration.IConfigurationSource> Sources { get { throw null; } }
+        public bool CollectChildKeysIndependently { get { throw null; } set { } }
```

Comment:
We can't add public APIs without going through the [API review process](https://github.com/dotnet/runtime/blob/main/docs/project/api-review-process.md)

Are these new APIs strictly necessary to fix the issue? They don't feel like they meet the public API bar, but instead it feels like internal implementation detail.

Reply:
Struggling how to optimize the performance without changing the behavior..